lib/trento/activity_logging/activity_logger.ex

```diff
@@ -60,4 +74,27 @@ defmodule Trento.ActivityLog.ActivityLogger do
         )
     end
   end
+
+  defp get_retention_time do
+    case Repo.one(Settings.base_query()) do
+      %Settings{} = settings -> {:ok, settings.retention_time}
+      nil -> {:error, :activity_log_settings_not_configured}
+    end
+  end
+
+  defp calculate_expiration_date(%RetentionTime{value: value, unit: :day}),
```

Comment:
Credo is complaining about pipelining only on function.

Here's an idea of how we could revisit this:
```elixir
  @day 24 * 60 * 60

  defp calculate_expiration_date(retention_time),
    do: DateTime.add(DateTime.utc_now(), -to_seconds(retention_time), :second)

  defp to_seconds(%RetentionTime{value: value, unit: :day}), do: value * @day
  defp to_seconds(%RetentionTime{value: value, unit: :week}), do: value * 7 * @day
  defp to_seconds(%RetentionTime{value: value, unit: :month}), do: value * 30 * @day
  defp to_seconds(%RetentionTime{value: value, unit: :year}), do: value * 365 * @day
```

Side note: we are assuming a month being always `30 days` and a year being always `365 days`.
I am fine with this approximation, just wondering whether worth making it explicit to users.

Reply:
The feedback from Credo is pretty straightforward, I fixed it in https://github.com/trento-project/web/pull/2792/commits/a2dc26e7ccaff974e1238595573be50c05d2ad09

Reply:
[This function](https://hexdocs.pm/elixir/1.15.7/DateTime.html#add/4) accepts time units as a parameter. I wonder if we can pass the retention time unit to this, and not have to approximate a month as 30 days?

Reply:
Afaik the only unit that intersects our needs is `:day`. We can indeed refactor in this sense, it may be more readable. Good point

Reply:
Done in https://github.com/trento-project/web/pull/2792/commits/ab7b9f27a7bd94b926adc60965ecbe0138b9f89e

Reply:
@balanza : There is also this one that looks promising: https://hexdocs.pm/ecto/Ecto.Query.API.html#datetime_add/3 and has all the units of interest.